subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/RelativePathTracker.java

```diff
@@ -60,13 +62,31 @@ public boolean isRoot() {
         return segments;
     }
 
+    public String toAbsolutePath() {
+        if (segments.isEmpty()) {
+            return rootName;
```

Comment:
I find it very strange to track the absolute path of the root as the root name.

Reply:
I found tracking the `rootName` like this very strange in general. I'm happy to take a look at this and figure out how we could write it better.